### ISSUE TITLE
Vsync causing network judder

### ISSUE THREAD

**Issue by Slashic** [moonlight-stream/moonlight-qt#1448]
**Describe the bug**
Only when vsync is enabled there is 0.40 to 0.90% frame dropped attributed to network jitter when the framerate of the game on the host can't cap the requested framerate and starts to engage adaptive sync on the client.
No jitter occurs if the framerate is stable and stays capped at the requested framerate in moonlight.
An occasional stutter/judder is visible but may be caused by the game itself or the host GPU being bottlenecked, I believe this is purely an error on the statistics display.

Does not happen at all with vsync off: tearing appears and an occasional stutter is still happening.

This seems similar to what is described in: https://github.com/moonlight-stream/moonlight-qt/issues/1323

**Steps to reproduce**
Launch a game through steam

**Screenshots**
N/A

**Affected games**
Metaphor Re:Fantazio, Ys 9, Vision of Mana...

**Other Moonlight clients**
Can't test that sorry

**Moonlight settings (please complete the following information)**
Use any of the codecs: H264/HEVC/AV1 (HDR enabled, 4:2:0)
Fullscreen or Windowed Fullscreen
Vsync ON
Any bitrate
4k 120fps or 60fps
Hardware AMD encoder & decoder (host & client)

**Does the problem still occur after reverting settings back to default?**
Yes (with vsync ON)

**Client PC details**
 - OS: Windows 10 22H2
 - Moonlight Version: v6.1.0
 - GPU: Radeon 6600

**Server PC details**
 - OS: Windows 10 22H2
 - Sunshine v2024.1020.30518
 - GPU: AMD Radeon RX 7900 XTX
 - GPU driver: 24.10.1

**Additional context**
This is on a 1Gbe Ethernet network

**Comment by cgutman** [comment]
This is by design. The encoding rate from the host is not (and cannot be) perfectly synchronized with the rate of the rendering on the client PC. If the host is producing slightly more frames than the client can display, the pacing logic has to drop one every so often to avoid video latency accumulating.

**Comment by Slashic** [comment]
Thanks for your reply, this is what i suspected too and indeed that seems logical. I was just a bit confused seeing this mechanism attributed to the network itself and not to the display sync or renderer, this could lead to false diagnostics if unaware.

**Comment by onelightinthedark** [comment]
i've been trying to diagnose an occasional problem with frames being dropped due to 'network jitter'.. does this mean that it could actual be a hardware/software issue on the client/host and not a problem with my network?

**Comment by Slashic** [comment]
Yes, that's what it seems to be, this metric isn't precise enough i guess.
